vlib:MSVC C++ Test Failing

### Describe the bug

When I try to compile `project_with_cpp_code` using MSVC on Windows, I get the following error.

### Reproduction Steps

It occurs on Windows with Visual Studio.
Run `v test -cc msvc vlib/v/tests/project_with_cpp_code` in the root folder of V.

### Expected Behavior

Errors like Current Behavior occur.

### Current Behavior

```
PS C:\v> v -cc msvc test vlib/v/tests/project_with_cpp_code
---- Testing... --------------------------------------------------------------------------------------------------------
 FAIL     0.000 ms vlib/v/tests/project_with_cpp_code/compiling_cpp_files_with_a_cplusplus_compiler_test.c.v
>> compilation failed:
C:\v\vlib\v\tests\project_with_cpp_code\implementation.obj not found, building it (with msvc)...
builder error: msvc: failed to build a thirdparty object after 0/5 retries, cmd: "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\bin\Hostx64\x64\cl.exe" /nologo /volatile:ms /MDd /D_DEBUG  -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt" -I "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared"  /c "C:\v\vlib\v\tests\project_with_cpp_code/implementation.c" /Fo"C:\v\vlib\v\tests\project_with_cpp_code\implementation.obj"

------------------------------------------------------------------------------------------------------------------------
Failed command 1:    "C:\v\v.exe" -skip-running -cc msvc  -b c  -o "C:\Users\keito\AppData\Local\Temp\v_0\tsession_6c5c_01JC083SCW27T63DGYHSWBV0JH\0_0\compiling_cpp_files_with_a_cplusplus_compiler_test.exe" "C:\v\vlib\v\tests\project_with_cpp_code\compiling_cpp_files_with_a_cplusplus_compiler_test.c.v"
Summary for all V _test.v files: 1 failed, 1 total. Elapsed time: 1678 ms, on 1 job. Comptime: 1504 ms. Runtime: 0 ms.
```

### Possible Solution

_No response_

### Additional Information/Context

When `compiling_cpp_files_with_a_cplusplus_compiler_test.c.v` is changed  `#flag @VMODROOT/implementation.o` to `#flag @VMODROOT/implementation.obj` I get the following error. (Please excuse the Japanese)

```
PS C:\v> v -cc msvc test vlib/v/tests/project_with_cpp_code
---- Testing... --------------------------------------------------------------------------------------------------------
 FAIL     0.000 ms vlib/v/tests/project_with_cpp_code/compiling_cpp_files_with_a_cplusplus_compiler_test.c.v
>> compilation failed:
compiling_cpp_files_with_a_cplusplus_compiler_test.exe.01JC08924ZY1PCQ8QC6GGG9Y1Z.tmp.c
LINK : fatal error LNK1104: ファイル 'C:\v\vlib\v\tests\project_with_cpp_code\implementation.obj' を開くことができません。

builder error: msvc error

------------------------------------------------------------------------------------------------------------------------
Failed command 1:    "C:\v\v.exe" -skip-running -cc msvc  -b c  -o "C:\Users\keito\AppData\Local\Temp\v_0\tsession_2314_01JC088ZK927Q8956971CKTCC4\0_1\compiling_cpp_files_with_a_cplusplus_compiler_test.exe" "C:\v\vlib\v\tests\project_with_cpp_code\compiling_cpp_files_with_a_cplusplus_compiler_test.c.v"
Summary for all V _test.v files: 1 failed, 1 total. Elapsed time: 3689 ms, on 1 job. Comptime: 3522 ms. Runtime: 0 ms.
```

### V version

latest

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.8 49513af.0585d96
OS: windows, Microsoft Windows 11 Pro v22631 64 �r�b�g
Processor: 16 cpus, 64bit, little endian,

getwd: C:\Users\keito
vexe: C:\v\v.exe
vexe mtime: 2024-11-06 08:02:11

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\keito\.vmodules
VTMP: OK, value: C:\Users\keito\AppData\Local\Temp\v_0

Git version: git version 2.45.2.windows.1
Git vroot status: weekly.2024.45-13-g0585d964
.git/config present: true

CC version: cc (Rev1, Built by MSYS2 project) 14.2.0
thirdparty/tcc status: thirdparty-windows-amd64 b425ac82
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzJiMjQ0OGU3Y2M1YTc4NTQyOGE5OWUiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.FfMfH3Du45S1ur2oJrm-Akn6DvMx2IYKuXPtiEDWMDE">Huly&reg;: <b>V_0.6-21219</b></a></sub>

--- a/cmd/tools/modules/testing/common.v
+++ b/cmd/tools/modules/testing/common.v
@@ -226,7 +226,6 @@ pub fn new_test_session(_vargs string, will_compile bool) TestSession {
 		skip_files << 'examples/coroutines/coroutines_bench.v'
 		$if msvc {
 			skip_files << 'vlib/v/tests/consts/const_comptime_eval_before_vinit_test.v' // _constructor used
-			skip_files << 'vlib/v/tests/project_with_cpp_code/compiling_cpp_files_with_a_cplusplus_compiler_test.c.v'
 		}
 		$if solaris {
 			skip_files << 'examples/pico/pico.v'

--- a/vlib/v/builder/msvc_windows.v
+++ b/vlib/v/builder/msvc_windows.v
@@ -401,7 +401,9 @@ fn (mut v Builder) build_thirdparty_obj_file_with_msvc(mod string, path string, 
 	}
 	println('${obj_path} not found, building it (with msvc)...')
 	flush_stdout()
-	cfile := '${path_without_o_postfix}.c'
+	cfile := cfile := if os.exists('${path_without_o_postfix}.c') { '${path_without_o_postfix}.c' } else {
+               '${path_without_o_postfix}.cpp'
+       }
 	flags := msvc_string_flags(moduleflags)
 	inc_dirs := flags.inc_paths.join(' ')
 	defines := flags.defines.join(' ')

--- a/vlib/v/builder/msvc_windows.v
+++ b/vlib/v/builder/msvc_windows.v
@@ -401,9 +401,11 @@ fn (mut v Builder) build_thirdparty_obj_file_with_msvc(mod string, path string, 
 	}
 	println('${obj_path} not found, building it (with msvc)...')
 	flush_stdout()
-	cfile := cfile := if os.exists('${path_without_o_postfix}.c') { '${path_without_o_postfix}.c' } else {
-               '${path_without_o_postfix}.cpp'
-       }
+	cfile := if os.exists('${path_without_o_postfix}.c') {
+		'${path_without_o_postfix}.c'
+	} else {
+		'${path_without_o_postfix}.cpp'
+	}
 	flags := msvc_string_flags(moduleflags)
 	inc_dirs := flags.inc_paths.join(' ')
 	defines := flags.defines.join(' ')

